data_adapter_oemof/build_datapackage.py

```diff
@@ -1,92 +1,185 @@
         parametrized_sequences = {}
-
-        for (process, data) in process_data.items():
-            # Getting Process parameters:
-            process_type: str = PROCESS_TYPE_MAP[process]
-            adapter = TYPE_MAP[process_type]
-
-            # Apply Adapter to every instance of the Process:
-            scalars = data.scalars.apply(
-                adapter.parametrize_dataclass,
-                struct=es_structure[process],
-                process_type=process_type,
+        foreign_keys = {}
+        for (process_name, struct) in es_structure.items():
+            process = adapter.get_process(process_name)
+            facade_adapter_name: str = PROCESS_TYPE_MAP[process_name]
+            facade_adapter = TYPE_MAP[facade_adapter_name]
+
+            scalars = process.scalars.apply(
+                func=facade_adapter.parametrize_dataclass,
+                struct=struct,
                 axis=1,
             )
-            # Maybe work on timeseries here:
-            sequences = data.timeseries
-
-            # Add process to parametrized_elements dict:
-            if process_type in parametrized_elements.keys():
-                parametrized_elements[process_type] = pd.concat(
-                    [
-                        pd.DataFrame([param.as_dict() for param in scalars.values]),
-                        parametrized_elements[process_type],
-                    ],
+
+            scalars = pd.DataFrame(scalars.to_list())
+            if "profiles" in facade_adapter.__dict__:
+                if len(facade_adapter.profiles) != 1:
+                    warnings.warn(
+                        message="Functionality to use more than one timeseries per process is not "
+                        "implemented yet"
+                    )
+
+                else:  # only one timeseries is implemented yet
+                    if not process.timeseries.empty:
+                        # Creating Sequences Dataframes
+                        if facade_adapter_name not in parametrized_sequences.keys():
+                            parametrized_sequences[
+                                facade_adapter_name
+                            ] = refactor_timeseries(timeseries=process.timeseries)
+
+                        else:
+                            parametrized_sequences[facade_adapter_name].update(
+                                refactor_timeseries(timeseries=process.timeseries)
+                            )
+
+                        # Adding profile column and naming to scalars:
+                        # Tabular will search for this
+                        column_name = facade_adapter.profiles[0]
+
+                        scalars[column_name] = (
+                                # Name of
+                                process.timeseries.columns.difference(core.TIMESERIES_COLUMNS.keys())[0]
+                                + "_"
+                                + scalars["region"]
+                        )
```

Comment:
My idea is do move this part to parametrize_dataclass and its mappings (therefore we have to introduce component specific mappings next to default mappings ad change behaviour of mapping a bit).
So that for example for a load component, there would exist a mapping something like this:
```
Default:
  amount: amount
LoadAdapter:
  profile: load
```
The mapper would see that there is no column `load` in scalars and would check if such column exists in timeseries, if so it would simply return the value "load" as this has to be set as profile.
Additionally, we should add a calculation to address region-based profiles (as you already enabled TS like "load-TH" - very good by the way!!!)